dsp/modules/llama.py

```diff
@@ -0,0 +1,93 @@
+try:
+    from llama_cpp import Llama
+except ImportError as exc:
+    raise ModuleNotFoundError(
+        "You need to install llama_cpp library to use gguf models."
+    ) from exc
+
+from dsp.modules.lm import LM
+from typing import Any, Literal
+
+
+class LlamaCpp(LM):
+    def __init__(
+            self,
+            model: str,
+            llama_model: Llama,
+            model_type: Literal["chat", "text"] = None,
+            **kwargs,
+    ):
+        super().__init__(model)
```

Comment:
so this adds a `"model": "llama"` kwarg to the model_kwargs from the `LM` class inheritance. I'm curious if this is expected LlamaCPP behavior since typically, this field is populated with the actual model name.

Reply:
worth noting in documentation given the difference from other LMs in DSPy